tests/integration/azure/test_locations.py

```diff
@@ -185,11 +203,51 @@ def test_overlapping_location(caplog, ws, sql_backend, inventory_schema):
     azurerm = AzureResources(azure_mgmt_client, graph_client)
 
     location_migration = ExternalLocationsMigration(
-        ws, location_crawler, AzureResourcePermissions(installation, ws, azurerm, location_crawler), azurerm
+        ws,
+        location_crawler,
+        AzureResourcePermissions(installation, ws, azurerm, location_crawler),
+        azurerm,
+        runtime_ctx.principal_acl,
     )
     try:
         leftover_loc_urls = location_migration.run()
         assert "abfss://uctest@ziyuanqintest.dfs.core.windows.net/" in leftover_loc_urls
         assert "overlaps with an existing external location" in caplog.text
     finally:
         save_delete_location(ws, "uctest_ziyuanqintest_overlap")
+
+
+def test_run_validate_acl(make_cluster_permissions, ws, make_user, make_cluster, az_cli_ctx, env_or_skip):
+
+    az_cli_ctx.with_dummy_resource_permission()
+    az_cli_ctx.save_locations()
+    cluster = make_cluster(single_node=True, spark_conf=_SPARK_CONF, data_security_mode=DataSecurityMode.NONE)
+    user = make_user()
+    make_cluster_permissions(
+        object_id=cluster.cluster_id,
+        permission_level=PermissionLevel.CAN_ATTACH_TO,
+        user_name=user.user_name,
+    )
+
+    location_migration = az_cli_ctx.azure_external_locations_migration
+    try:
+        location_migration.run()
+        permissions = ws.grants.get(
+            SecurableType.EXTERNAL_LOCATION, env_or_skip("TEST_A_LOCATION"), principal=user.user_name
+        )
+        expected_permission = PrivilegeAssignment(
+            principal=user.user_name,
+            privileges=[Privilege.CREATE_EXTERNAL_TABLE, Privilege.CREATE_EXTERNAL_VOLUME, Privilege.READ_FILES],
+        )
+        assert expected_permission in permissions.privilege_assignments
```

Comment:
Does this work?...

Reply:
@nfx I need access to azure labs mi access connector read only